models/amity.py

```diff
@@ -0,0 +1,428 @@
+        self.allocated_people = []
+        self.unallocated_people = []
+        self.fellows = []
+        self.allocated_fellows = []
+        self.staff = []
+        self.allocated_staff = []
+
+    def create_room(self, args):
+        """Create new room(s)"""
+        print spacer
+        new_rooms = []
+        for room in args["<room_name>"]:
+            if room.lower() in [r.name.lower() for r in self.rooms]:
+                print "One or more rooms you tried to create already exist! " \
+                    "Please try again."
+                print spacer
+                return
+            if args["Office"]:
+                new_room = Office(room)
+                self.offices.append(new_room)
+                self.check_vacant_offices()
+            elif args["Living"]:
+                new_room = Living(room)
+                self.livingspaces.append(new_room)
+                self.check_vacant_livingspaces()
+            self.rooms.append(new_room)
+            new_rooms.append(new_room)
+        print "You have successfully added the following rooom(s):"
+        for new_room_ in new_rooms:
+            print "Name: " + ''.join(new_room_.name) + " | Type: " \
+                + new_room_.room_type
+        print spacer
+
+    def create_room_from_db(self, args):
+        """Create new room(s) from existing database"""
+        print spacer
+        new_rooms = []
+        for room in args["<room_name>"]:
+            if room.lower() in [r.name.lower() for r in self.rooms]:
+                print "One or more rooms you tried to create already exist! " \
+                    "Please try again."
+                print spacer
+                return
+            if args["Office"]:
+                new_room = Office(room)
+                self.offices.append(new_room)
+                self.check_vacant_offices()
+            elif args["Living"]:
+                new_room = Living(room)
+                self.livingspaces.append(new_room)
+                self.check_vacant_livingspaces()
+            self.rooms.append(new_room)
+            new_rooms.append(new_room)
+        print "You have successfully added the following rooom(s):"
+        for new_room_ in new_rooms:
+            print "Name: " + ''.join(new_room_.name) + " | Type: " \
+                + new_room_.room_type
+        print spacer
+
+    def check_vacant_offices(self):
+        """Add vacant offices to lists; remove full ones from lists"""
+        for office in self.offices:
+            if len(office.occupants) < office.capacity:
+                if office not in self.vacant_offices:
+                    self.vacant_offices.append(office)
+                    self.vacant_rooms.append(office)
+            elif len(office.occupants) >= office.capacity:
+                if office in self.vacant_offices:
+                    self.vacant_offices.remove(office)
+                    self.vacant_rooms.remove(office)
+
+    def check_vacant_livingspaces(self):
```

Comment:
**Code Quality: Nitpick**

Method definition here seems very similar to the one before. See if it's possible to compress into a single method definition. Take advantage of the fact that objects are reference types and once the reference is modified the changes reflect wherever the instances are used.